services/graphql/src/mongoose/schema/lineitem.js

```diff
@@ -240,11 +240,9 @@ schema.pre('save', async function updateRelatedModels() {
 schema.pre('save', async function updateEvents() {
   if (this.isModified('orderId')) {
     const { advertiserId, orderId } = this;
-    ['view', 'click'].forEach((modelName) => {
-      connection.model(modelName).updateMany({ lineitemId: this._id }, {
-        $set: { advertiserId, orderId },
-      }).catch(e => logError(e));
-    });
+    connection.model('event').updateMany({ lineitemId: this._id }, {
```

Comment:
This query must also include the proper types, e.g:
`{ lineitemId: this._id, type: { $in: ['view', 'click'] } }`

Reply:
Index this as well